mocket/mocket.py

```
@@ -64,8 +64,15 @@ def collect(cls, data) -> None:
     @classmethod
     def reset(cls) -> None:
         for r_fd, w_fd in cls._socket_pairs.values():
-            os.close(r_fd)
-            os.close(w_fd)
+            try:
+                os.close(r_fd)
+            except OSError:
+                pass
```

I assume overall we're defending against the socket already being closed? If so, is there a more specific execption we can catch to avoid silently failing if say there's a permissions error?

```py
try:
    os.close(r_fd)
except OSError as ex:
    if not ex.is_closed: # changeme
        raise
```

---

I did run into this issue when doing a full run of recordings with our app, tbh I did not look into this deeper. Let's keep this on the side for now, I don't have enough data to provide a more detailed description to the maintainers.